recipes/Time_Series/Few-shot_Finetuning_and_Evaluation.ipynb

```diff
@@ -0,0 +1,891 @@
+    "\n",
+    "# Count params\n",
+    "print(\n",
+    "    \"Number of params after freezing the backbone\",\n",
+    "    count_parameters(finetune_forecast_model),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set up a Trainer for Finetune Few-shot 5%\n",
+    "\n",
+    "Configure a Trainer for use in fine-tuning and evaluating the model. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Important parameters\n",
+    "learning_rate = 0.001\n",
+    "num_epochs = 1  # Ideally, we need more epochs (try offline preferably in a gpu for faster computation)\n",
+    "batch_size = 64\n",
+    "\n",
+    "print(f\"Using learning rate = {learning_rate}\")\n",
+    "finetune_forecast_args = TrainingArguments(\n",
+    "    output_dir=os.path.join(OUT_DIR, \"output\"),\n",
+    "    overwrite_output_dir=True,\n",
+    "    learning_rate=learning_rate,\n",
+    "    num_train_epochs=num_epochs,\n",
+    "    do_eval=True,\n",
+    "    eval_strategy=\"epoch\",\n",
+    "    per_device_train_batch_size=batch_size,\n",
+    "    per_device_eval_batch_size=batch_size,\n",
+    "    dataloader_num_workers=8,\n",
+    "    report_to=None,\n",
+    "    save_strategy=\"epoch\",\n",
+    "    logging_strategy=\"epoch\",\n",
+    "    save_total_limit=1,\n",
+    "    logging_dir=os.path.join(OUT_DIR, \"logs\"),  # Make sure to specify a logging directory\n",
+    "    load_best_model_at_end=True,  # Load the best model when training ends\n",
+    "    metric_for_best_model=\"eval_loss\",  # Metric to monitor for early stopping\n",
+    "    greater_is_better=False,  # For loss\n",
+    ")\n",
+    "\n",
+    "# Create the early stopping callback\n",
+    "early_stopping_callback = EarlyStoppingCallback(\n",
+    "    early_stopping_patience=10,  # Number of epochs with no improvement after which to stop\n",
+    "    early_stopping_threshold=0.0,  # Minimum improvement required to consider as improvement\n",
+    ")\n",
+    "tracking_callback = TrackingCallback()\n",
+    "\n",
+    "# Optimizer and scheduler\n",
+    "optimizer = AdamW(finetune_forecast_model.parameters(), lr=learning_rate)\n",
+    "scheduler = OneCycleLR(\n",
+    "    optimizer,\n",
+    "    learning_rate,\n",
+    "    epochs=num_epochs,\n",
+    "    steps_per_epoch=math.ceil(len(train_dataset) / (batch_size)),\n",
+    ")\n",
+    "\n",
+    "finetune_forecast_trainer = Trainer(\n",
+    "    model=finetune_forecast_model,\n",
+    "    args=finetune_forecast_args,\n",
+    "    train_dataset=train_dataset,\n",
+    "    eval_dataset=valid_dataset,\n",
+    "    callbacks=[early_stopping_callback, tracking_callback],\n",
+    "    optimizers=(optimizer, scheduler),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Train the Model\n",
+    "\n",
+    "This takes about 2 minutes."
```

Comment:
@wgifford could you provide a description of what is happening under the hood with fine-tuning? Is this creating an adapter? Or is it coming up with a few-shot prompt?